daft/execution/actor_pool_udf.py

```diff
@@ -14,39 +16,95 @@
 
 logger = logging.getLogger(__name__)
 
+_SENTINEL = ("__EXIT__", 0)
 
-def actor_event_loop(uninitialized_projection: ExpressionsProjection, conn: Connection) -> None:
-    """Event loop that runs in a actor process and receives MicroPartitions to evaluate with an initialized UDF projection.
 
-    Terminates once it receives None.
-    """
-    initialized_projection = ExpressionsProjection([e._initialize_udfs() for e in uninitialized_projection])
+class SharedMemoryTransport:
+    def write(self, data: bytes) -> tuple[str, int]:
+        shm = shared_memory.SharedMemory(create=True, size=len(data))
```

Comment:
Getting some weird
```
/usr/lib/python3.12/multiprocessing/resource_tracker.py:267: UserWarning: resource_tracker: '/psm_97f6afa9': [Errno 2] No such file or directory: '/psm_97f6afa9'
  warnings.warn('resource_tracker: %r: %s' % (name, e))
  ```
  errors, probably from double unlinking (we do it manually but python's resource tracker is also trying to manage lifetimes). We can do something like
```
resource_tracker.unregister(shm._name, "shared_memory")
```
and import `from multiprocessing import resource_tracker` at the top to avoid the errors